server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -85,20 +85,38 @@
   private static final Interner<DataSegment> DATA_SEGMENT_INTERNER = Interners.newWeakInterner();
   private static final EmittingLogger log = new EmittingLogger(SQLMetadataSegmentManager.class);
 
-  // Use to synchronize start() and stop(). These methods should be synchronized to prevent from being called at the
-  // same time if two different threads are calling them. This might be possible if a druid coordinator gets and drops
-  // leadership repeatedly in quick succession.
-  private final Object lock = new Object();
+  /**
+   * Use to synchronize {@link #start()}, {@link #stop()}, {@link #poll()}, and {@link #isStarted()}. These methods
+   * should be synchronized to prevent from being called at the same time if two different threads are calling them.
+   * This might be possible if a druid coordinator gets and drops leadership repeatedly in quick succession.
+   */
+  private final ReentrantReadWriteLock readWriteLock = new ReentrantReadWriteLock();
```

Comment:
I wonder if this should be the responsibility of the Lifecycle to synchronize start/stop of objects ?